rpc/trace_test.go

```diff
@@ -11,30 +11,63 @@ import (
 	"github.com/test-go/testify/require"
 )
 
-// TestNotImplemented checks the method not yet implemented
-func TestNotImplemented(t *testing.T) {
+// TestTraceTransaction tests the TransactionTrace method
+func TestTransactionTrace(t *testing.T) {
 	testConfig := beforeEach(t)
 
+	var expectedResp InvokeTxnTrace
+	if testEnv == "mock" {
+		var rawjson struct {
+			Result InvokeTxnTrace `json:"result"`
+		}
+		expectedrespRaw, err := os.ReadFile("./tests/0xff66e14fc6a96f3289203690f5f876cb4b608868e8549b5f6a90a21d4d6329.json")
+		require.NoError(t, err, "Error ReadFile for TestTraceTransaction")
+
+		err = json.Unmarshal(expectedrespRaw, &rawjson)
+		require.NoError(t, err, "Error unmarshalling testdata TestTraceTransaction")
+	
+		txnTrace, err := json.Marshal(rawjson.Result)
+		require.NoError(t, err, "Error unmarshalling testdata TestTraceTransaction")
+		err = json.Unmarshal(txnTrace, &expectedResp)
+	}
+
 	type testSetType struct {
-		MissingMethod string
+		TransactionHash *felt.Felt
+		ExpectedResp    TxnTrace
+		ExpectedError   *RPCError
 	}
 	testSet := map[string][]testSetType{
-		"devnet": {},
-		"mainnet": {
-			{MissingMethod: "starknet_traceTransaction"},
-		},
-		"mock": {},
-		"testnet": {
-			{MissingMethod: "starknet_traceTransaction"},
+		"mock": {
+			testSetType{
+				TransactionHash: utils.TestHexToFelt(t, "0xff66e14fc6a96f3289203690f5f876cb4b608868e8549b5f6a90a21d4d6329"),
+				ExpectedResp:    expectedResp,
+				ExpectedError:   nil,
+			},
+			testSetType{
+				TransactionHash: utils.TestHexToFelt(t, "0xc0ffee"),
+				ExpectedResp:    nil,
+				ExpectedError:   ErrInvalidTxnHash,
+			},
+			testSetType{
+				TransactionHash: utils.TestHexToFelt(t, "0xf00d"),
+				ExpectedResp:    nil,
+				ExpectedError:   &RPCError{
+					code: 10,
+					message: "No trace available for transaction",
+					data: TransactionRejected,
+				},
+			},
 		},
+		"devnet":  {},
+		"mainnet": {},
 	}[testEnv]
 
 	for _, test := range testSet {
-		var out string
-		err := do(context.Background(), testConfig.provider.c, test.MissingMethod, &out)
-
-		if err == nil || err.Error() != "Method Not Found" {
-			t.Fatalf("Method %s is now available, got %v\n", test.MissingMethod, err)
+		resp, err := testConfig.provider.TransactionTrace(context.Background(), test.TransactionHash)
+		if err != nil {
+			require.Equal(t, test.ExpectedError, err)
+		} else {
```

Comment:
Might be neater without the else

Reply:
This is funny! I pointed out the opposite over here https://github.com/NethermindEth/starknet.go/pull/332#discussion_r1327199608

Reply:
Lol, I'm happy either way. Guess we all have our own styles :)